src/metpy/calc/thermo.py

```diff
@@ -3659,3 +3659,86 @@ def cross_totals(pressure, temperature, dewpoint):
 
     # Calculate vertical totals.
     return td850 - t500
+
+
+@exporter.export
+@preprocess_and_wrap()
+@check_units('[pressure]', '[temperature]', '[temperature]', '[speed]')
+def sweat_index(pressure, temperature, dewpoint, speed, direction):
+    """Calculate SWEAT Index.
+
+    SWEAT Index derived from [Miller1972]:
+
+    .. math:: SWEAT = 12Td850 + 20(TT - 49) + 2f850 + f500 + 125(S + 0.2)
+
+    where:
+
+    * :math:`Td850` is the dewpoint at 850 hPa, the first term is set to zero
+       if Td850 is negative
+    * :math:`TT` is the total totals index, the second term is set to zero
+       if TT is less than 49
+    * :math:`f850` is the wind speed at 850 hPa
+    * :math:`f500` is the wind speed at 500 hPa
+    * :math:`S` is the shear term: :math:`sin(the wind direction at 850 hPa - the wind direction at 500 hPa)`.
+       It is set to zero if any of the following conditions are not met:
+       1. the wind direction at 850 hPa is between 130 - 250 degrees
+       2. the wind direction at 500 hPa is between 210 - 310 degrees
+       3. (the wind direction at 500 - the wind direction at 850 hPa)
+           is greater than zero
+       4. both the wind speeds are greater than or equal to 15 kts
```

Comment:
This cleans up the formatting of the docs a bunch and looks good when rendered to HTML:
```suggestion
    .. math:: SWEAT = 12Td_{850} + 20(TT - 49) + 2f_{850} + f_{500} + 125(S + 0.2)

    where:

    * :math:`Td_{850}` is the dewpoint at 850 hPa; the first term is set to zero
      if :math:`Td_{850}` is negative.
    * :math:`TT` is the total totals index; the second term is set to zero
      if :math:`TT` is less than 49
    * :math:`f_{850}` is the wind speed at 850 hPa
    * :math:`f_{500}` is the wind speed at 500 hPa
    * :math:`S` is the shear term: :math:`\sin{(dd_{850} - dd_{500})}`, where
      :math:`dd_{850}` and :math:`dd_{500}` are the wind directions at 850 hPa and 500 hPa,
      respectively. It is set to zero if any of the following conditions are not met:

    1. :math:`dd_{850}` is between 130 - 250 degrees
    2. :math:`dd_{500}` is between 210 - 310 degrees
    3. :math:`dd_{500} - dd_{850} > 0`
    4. both the wind speeds are greater than or equal to 15 kts
```